worlds/seaofthieves/__init__.py

```diff
@@ -0,0 +1,192 @@
+        self.clientInputs.sotOptionsDerived = self.sotOptionsDerived
+        self.clientInputs.regionRules = self.region_rules
+        self.clientInputs.shopWarehouse = self.locationCollection.shops
+        self.clientInputs.shopWarehouse.remove_non_pickle_members()
+        self.clientInputs.hintLibrary = self.buildHintLibrary().getStringLibrary(self.random)
+        self.clientInputs.multiworldHints = MultiworldHints(self.multiworld, self.player, self.random, 50)
+        self.clientInputs.to_file(output_file_and_directory)
+
+    def pre_fill_seals(self) -> int:
+
+        # right now we just have seals, so this works, but it wont soon
+
+        seal_items = [
+            SOTItem(Items.seal_gh.name, Items.seal_gh.classification, Items.seal_gh.id, self.player),
+            SOTItem(Items.seal_ma.name, Items.seal_ma.classification, Items.seal_ma.id, self.player),
+            SOTItem(Items.seal_af.name, Items.seal_af.classification, Items.seal_af.id, self.player),
+            SOTItem(Items.seal_rb.name, Items.seal_rb.classification, Items.seal_rb.id, self.player),
+            SOTItem(Items.seal_oos.name, Items.seal_oos.classification, Items.seal_oos.id, self.player)
+        ]
+        seal_locations = [
+            self.multiworld.get_location(Seals.Seals.L_VOYAGE_COMP_GH_TOTAL, self.player),
+            self.multiworld.get_location(Seals.Seals.L_VOYAGE_COMP_MA_TOTAL, self.player),
+            self.multiworld.get_location(Seals.Seals.L_VOYAGE_COMP_OOS_TOTAL, self.player),
+            self.multiworld.get_location(Seals.Seals.L_VOYAGE_COMP_AF_TOTAL, self.player),
+            self.multiworld.get_location(Seals.Seals.L_VOYAGE_COMP_RB_TOTAL, self.player)
+        ]
+
+        for i in seal_items:
+            self.item_name_to_id[i.name] = i.code
+
+        all_state = self.multiworld.get_all_state(use_cache=True)
+
+        self.random.shuffle(seal_locations)
+
+        fill_restrictive(self.multiworld, all_state, seal_locations, seal_items, True, lock=True,
+                         name="SOT Seals")
+
+        fod_location = self.multiworld.get_location(QuestMenu.MenuQuestAll.L_PIRATE_FOD, self.player)
+        self.item_name_to_id[Items.pirate_legend.name] = Items.pirate_legend.id
+        itm = self.create_item(Items.pirate_legend.name)
+        fill_restrictive(self.multiworld, all_state, [fod_location], [itm], True, lock=True,
+                         name="SOT Seals")
+
+        return 5 + 1
+
+    def buildHintLibrary(self) -> HintLibrary:
+        hl = HintLibrary()
+        for location in self.multiworld.get_locations(self.player):
+            hint: Hint = hint_from_location(location)
+            classification: HintLibrary.Type = HintLibrary.Type.PROGRESSIVE
+            if location.item.classification == ItemClassification.filler:
+                classification = HintLibrary.Type.FILLER
+            elif location.item.classification == ItemClassification.trap:
+                classification = HintLibrary.Type.TRAP
```

Comment:
This looks like it'll treat Useful items as Progression (or Progressive as you put it). Not sure if this is intentional.